cachi2/core/package_managers/pip.py

```diff
@@ -273,74 +273,61 @@ def _generate_purl_dependency(package: dict[str, Any]) -> str:
     return purl.to_string()
 
 
-def _get_pip_metadata(package_dir: RootedPath) -> tuple[str, Optional[str]]:
-    """
-    Attempt to get the name and version of a Pip package.
+def _infer_package_name_from_origin_url(package_dir: RootedPath) -> str:
+    try:
+        repo_id = get_repo_id(package_dir.root)
+    except UnsupportedFeature:
+        raise PackageRejected(
+            reason="Unable to infer package name from origin URL",
+            solution=(
+                "Provide valid metadata in the package files or ensure"
+                "the git repository has an 'origin' remote with a valid URL."
+            ),
+            docs=PIP_METADATA_DOC,
+        )
 
-    First, try to parse the setup.py script (if present) and extract name and version
-    from keyword arguments to the setuptools.setup() call. If either name or version
-    could not be resolved and there is a setup.cfg file, try to fill in the missing
-    values from metadata.name and metadata.version in the .cfg file.
+    repo_name = Path(repo_id.parsed_origin_url.path.removesuffix(".git")).name
+    subpath = package_dir.subpath_from_root
+    resolved_name = Path(repo_name).joinpath(subpath)
+    return canonicalize_name(str(resolved_name).replace("/", "-")).strip("-.")
```

Comment:
I would like to not rearrange the commits in this PR at all + don't waste too much time here. Each commit would require modification in unit tests, and that is not trivial based on the extremely complicated logic in there. I chose a much easier path, so, delete everything and write this from scratch. 

I know, in GitHub the PR is basically unreviewable. Sorry for that.

Reply:
> I know, in GitHub the PR is basically unreviewable. Sorry for that.

First things first, GH has nothing to do with this, I review stuff locally with 50 lines of context usually and even with flags to git to detect moved code doesn't change the situation in any way.

> I would like to not rearrange the commits in this PR at all + don't waste too much time here. Each commit would require modification in unit tests, and that is not trivial based on the extremely complicated logic in there. I chose a much easier path, so, delete everything and write this from scratch.

The only blocker for splitting here is the unit tests here, I agree. But given that this refactor actually changes the behavioral logic such that we don't return the same results as previously from the `pip_metadata` helper function. Additionally, the tests that you introduce actually don't test the compound cases which is the whole point of the pip metadata querying logic (e.g. name comes from one project file, version from another one). Since it looks like you'll have to rework the unit tests quite a bit that gives you the opportunity to split the changes into commits - I have actually tried myself and was pretty straightforward except for the tests, but like I said, looks like you'll have to rework that.